src/ros_filter.cpp

```diff
@@ -2006,149 +2006,26 @@ void RosFilter<T>::periodicUpdate()
-           * We have a transform from map_frame_id_->base_link_frame_id_, but
-           * it would actually transform a given pose from
-           * base_link_frame_id_->map_frame_id_. We then used lookupTransform,
-           * whose first two arguments are target frame and source frame, to
-           * get a transform from base_link_frame_id_->odom_frame_id_.
-           * However, this transform would actually transform data from
-           * odom_frame_id_->base_link_frame_id_. Now imagine that we have a
-           * position in the map_frame_id_ frame. First, we multiply it by the
-           * inverse of the map_frame_id_->baseLinkFrameId, which will
-           * transform that data from map_frame_id_ to base_link_frame_id_.
-           * Now we want to go from base_link_frame_id_->odom_frame_id_, but
-           * the transform we have takes data from
-           * odom_frame_id_->base_link_frame_id_, so we need its inverse as
-           * well. We have now transformed our data from map_frame_id_ to
-           * odom_frame_id_. However, if we want other users to be able to do
-           * the same, we need to broadcast the inverse of that entire
-           * transform.
-           */
-          tf2::Transform map_odom_trans;
-          map_odom_trans.mult(world_base_link_trans, base_link_odom_trans);
-
-          geometry_msgs::msg::TransformStamped map_odom_trans_msg;
-          map_odom_trans_msg.transform = tf2::toMsg(map_odom_trans);
-          map_odom_trans_msg.header.stamp =
-            static_cast<rclcpp::Time>(filtered_position->header.stamp) + tf_time_offset_;
-          map_odom_trans_msg.header.frame_id = map_frame_id_;
-          map_odom_trans_msg.child_frame_id = odom_frame_id_;
-
-          world_transform_broadcaster_->sendTransform(map_odom_trans_msg);
-        } catch (...) {
-          // ROS_ERROR_STREAM_DELAYED_THROTTLE(5.0, "Could not obtain
-          // transform from "
-          //                                  << odom_frame_id_ << "->" <<
-          //                                  base_link_frame_id_);
-        }
-      } else {
-        std::cerr << "Odometry message frame_id was " <<
-          filtered_position->header.frame_id << ", expected " <<
-          map_frame_id_ << " or " << odom_frame_id_ << "\n";
-      }
-    }
-
-    // Retain the last published stamp so we can detect repeated transforms in future cycles
-    last_published_stamp_ = filtered_position->header.stamp;
-
-    // Fire off the position and the transform
-    if (!corrected_data) {
-      position_pub_->publish(std::move(filtered_position));
-    }
+  updateFilterWithMeasurements(cur_time);
 
-    if (print_diagnostics_) {
-      freq_diag_->tick();
+  // Get latest state and publish it
+  {
+    nav_msgs::msg::Odometry filtered_position;
+    if (getFilteredOdometryMessage(filtered_position)) {
+      corrected_data = publishPositionWithOdometry(std::move(filtered_position));
```

Comment:
If you make this method take a const ref, there's no need for the `std::move`. Not sure what that does in the context of a value parameter anyway.

Reply:
Fixed.